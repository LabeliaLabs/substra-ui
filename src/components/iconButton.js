import React from 'react';
import styled from '@emotion/styled';
import PropTypes from '../utils/propTypes';
import {ice} from '../variables/colors';

export const RoundButton = styled.button`
    display: inline-flex;
    align-items: center;
    justify-content: center;
    width: 30px;
    height: 30px;
    border-radius: 15px;
    border: 1px solid ${ice};
    padding: 0;
    background: none;
    cursor: pointer;
    
    &:hover {
        background-color: ${ice};
        transition: background-color 200ms ease-out;
    }
`;

export const IconButton = ({Icon, iconSize, ...props}) => (
    <RoundButton {...props}>
        <Icon width={iconSize} height={iconSize} />
    </RoundButton>
);

IconButton.propTypes = {
    Icon: PropTypes.component.isRequired,
    iconSize: PropTypes.number,
};

IconButton.defaultProps = {
    iconSize: 15,
};
