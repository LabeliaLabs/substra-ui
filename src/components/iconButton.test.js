import React from 'react';
import renderer from 'react-test-renderer';
import {cleanup} from 'react-testing-library';
import {IconButton} from './iconButton';
import Book from '../icons/book';

afterEach(cleanup);

test('It should render the same', () => {
    const component = renderer.create(<IconButton Icon={Book} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
