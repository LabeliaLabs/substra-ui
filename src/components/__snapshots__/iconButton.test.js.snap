// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render the same 1`] = `
.emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 30px;
  height: 30px;
  border-radius: 15px;
  border: 1px solid #e7e8e8;
  padding: 0;
  background: none;
  cursor: pointer;
}

.emotion-0:hover {
  background-color: #e7e8e8;
  -webkit-transition: background-color 200ms ease-out;
  transition: background-color 200ms ease-out;
}

<button
  className="emotion-0 emotion-1"
>
  <svg
    className=""
    height={15}
    viewBox="0 0 24 24"
    width={15}
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M19 16V3H6.5A1.5 1.5 0 0 0 5 4.5v11.837A3.486 3.486 0 0 1 6.5 16H19zm0 2H6.5a1.5 1.5 0 0 0 0 3H19v-3zM6.5 1H20a1 1 0 0 1 1 1v20a1 1 0 0 1-1 1H6.5A3.5 3.5 0 0 1 3 19.5v-15A3.5 3.5 0 0 1 6.5 1z"
      fill="#4B6073"
      fillRule="nonzero"
    />
  </svg>
</button>
`;
